Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -53,8 +53,11 @@ def check_xcode_up_to_date
         return unless MacOS::Xcode.installed?
         return unless MacOS::Xcode.outdated?
 
-        # CI images are going to end up outdated so don't complain.
-        return if ENV["TRAVIS"] || ENV["CIRCLECI"]
+        # Travis CI images are going to end up outdated so don't complain when
+        # `brew test-bot` runs `brew doctor` in the CI for the Homebrew/brew
+        # repository. This only needs to support whatever CI provider
+        # Homebrew/brew is currently using.
+        return if ENV["TRAVIS"]
```

Comment:
Function name is `check_xcode_up_to_date`, so I would add ```return unless OS.mac?``` in the very beginning :)

Anyways, related commit is:

```patch
$ git show 6ac0de1b
commit 6ac0de1b40699042ea124813f2e953285efbd05f
Author: Mike McQuaid <mike@mikemcquaid.com>
Date:   Fri Feb 17 21:27:40 2017 +0000

    Revert "Skip Xcode outdated check on CircleCI"

diff --git a/Library/Homebrew/extend/os/mac/diagnostic.rb b/Library/Homebrew/extend/os/mac/diagnostic.rb
index 3e0c1d0..6d30cd2 100644
--- a/Library/Homebrew/extend/os/mac/diagnostic.rb
+++ b/Library/Homebrew/extend/os/mac/diagnostic.rb
@@ -53,8 +53,8 @@ module Homebrew
         return unless MacOS::Xcode.installed?
         return unless MacOS::Xcode.outdated?
 
-        # CI images are going to end up outdated so don't complain.
-        return if ENV["TRAVIS"] || ENV["CIRCLECI"]
+        # Travis CI images are going to end up outdated so don't complain.
+        return if ENV["TRAVIS"]
```